src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/Cache.java

```diff
@@ -73,6 +72,24 @@ public int size() {
         return entries.size();
     }
 
+    public void setExpireDuration(final int duration, final TimeUnit unit) {
+        this.expireAfterNanos = unit.toNanos(duration);
+    }
+
+    public Stat stats() {
+        final List<Long> durations = new ArrayList<>();
+        if (entries.size() > 0) {
+            for (final Entry<V> e : entries.values()) {
```

Comment:
Stats are good, but the number of entries in the cache can be potentially big, don't they? Do we really care about this stats to pay the price?

Reply:
Or at least, given that this information is exclusively for logging, I think the whole stats code should be no-op unless a given logging level is set.

At any rate, this seems to be an unrelated matter and should be put in a separated PR. Do you agree?

Reply:
Agree. Another idea would be to add a button on a GUI to ask for stats. What do you think ?

Reply:
Ok for me. But please, in a separated PR.

Reply:
done